php7.2-stretch/Dockerfile

```diff
@@ -0,0 +1,174 @@
+FROM php:7.2-fpm-stretch AS build
+
+ENV PATH=/root/.local/bin:$PATH \
+    PROCFILE_PATH=/app/Procfile \
+    LC_ALL=C.UTF-8 \
+    NVM_DIR=/opt/nvm \
+    NVM_VERSION=0.34.0
+
+USER root
+
+RUN echo "force-unsafe-io" > /etc/dpkg/dpkg.cfg.d/02apt-speedup
+RUN echo "Acquire::http {No-Cache=True;};" > /etc/apt/apt.conf.d/no-cache
+
+RUN DEBIAN_FRONTEND=noninteractive dpkg-reconfigure -f noninteractive tzdata
+RUN apt-get update && apt-get upgrade -y
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -yqq \
+      apt-transport-https \
+      apt-utils \
+      build-essential \
+      curl \
+      dumb-init \
+      gnupg2 \
+      libc-client-dev \
+      nginx \
+      openssh-client \
+      unzip \
+      zlib1g-dev \
+      --no-install-recommends \
+  && rm -rf /var/lib/apt/lists/*
+
+ENV buildDependencies=" \
+    libbz2-dev \
+    libsasl2-dev \
+    "
+
+ENV runtimeDependencies=" \
+    imagemagick \
+    libfreetype6-dev \
+    libicu-dev \
+    libjpeg62-turbo-dev \
+    libjpeg-dev \
+    libkrb5-dev \
+    libldap2-dev \
+    libmagickwand-dev \
+    libmemcached-dev \
+    libmemcachedutil2 \
+    libpng-dev \
+    libpq-dev \
+    libssl-dev \
+    libwebp-dev \
+    libxml2-dev \
+    libxpm-dev \
+    libzip-dev \
+    "
+
+ENV extensions=" \
+    bcmath \
+    bz2 \
+    calendar \
+    exif \
+    iconv \
+    intl \
+    mbstring \
+    opcache \
+    pcntl \
+    pdo_pgsql \
+    pgsql \
+    soap \
+    xml \
+    xmlrpc \
+    zip \
+    "
+
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get -yqq install $buildDependencies --no-install-recommends \
```

Comment:
The `apt-get update` should not be needed.

Reply:
It was necessary because the cache was deleted previously but you are right, that this should not happen here but later and I removed it